src/Microsoft.AspNet.Mvc.TagHelpers/Internal/GlobbingUrlBuilder.cs

```diff
@@ -20,6 +20,10 @@ public class GlobbingUrlBuilder
     {
         private static readonly char[] PatternSeparator = new[] { ',' };
 
+        // Valid whitespace characters defined by the HTML5 spec.
+        private static readonly char[] ValidAttributeWhitespaceChars =
+            new[] { '\t', '\n', '\u000C', '\r', ' ' };
```

Comment:
(Especially because what's valid in one place might not be valid elsewhere, and that makes code more fragile. No easy answer to this aside from don't do it.)